llvm/test/CodeGen/RISCV/GlobalISel/legalizer/legalize-fptoi-rv32.mir

```diff
@@ -166,13 +265,27 @@ body:             |
   bb.1:
     liveins: $f10_d
 
-    ; CHECK-LABEL: name: fptosi_s1_s64
-    ; CHECK: liveins: $f10_d
-    ; CHECK-NEXT: {{  $}}
-    ; CHECK-NEXT: [[COPY:%[0-9]+]]:_(s64) = COPY $f10_d
-    ; CHECK-NEXT: [[FPTOSI:%[0-9]+]]:_(s32) = G_FPTOSI [[COPY]](s64)
-    ; CHECK-NEXT: $x10 = COPY [[FPTOSI]](s32)
-    ; CHECK-NEXT: PseudoRET implicit $x10
+    ; CHECK-FLOAT-LABEL: name: fptosi_s1_s64
+    ; CHECK-FLOAT: liveins: $f10_d
+    ; CHECK-FLOAT-NEXT: {{  $}}
+    ; CHECK-FLOAT-NEXT: [[COPY:%[0-9]+]]:_(s64) = COPY $f10_d
+    ; CHECK-FLOAT-NEXT: [[FPTOSI:%[0-9]+]]:_(s32) = G_FPTOSI [[COPY]](s64)
+    ; CHECK-FLOAT-NEXT: $x10 = COPY [[FPTOSI]](s32)
+    ; CHECK-FLOAT-NEXT: PseudoRET implicit $x10
+    ;
+    ; CHECK-NOFLOAT-LABEL: name: fptosi_s1_s64
+    ; CHECK-NOFLOAT: liveins: $f10_d
+    ; CHECK-NOFLOAT-NEXT: {{  $}}
+    ; CHECK-NOFLOAT-NEXT: [[COPY:%[0-9]+]]:_(s64) = COPY $f10_d
```

Comment:
Something is broken here. We're passing f10_d, but we don't have floating point?

Reply:
So f10_d is a register only available when D and F are enabled. So I should be using for example x10 as a livein register right?

Reply:
For rv32, you would need to use x10 and x11 to hold all 64 bits. I recommend writing an IR test and using -stop-before=legalizer to get the MIR